packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,545 @@
+    /// on-call team needed more time. Ultimately, we rejected all of them in favour of better on-call processes.
+    function configureTimelockGuard(uint256 _timelockDelay) external {
+        // Record the calling Safe
+        Safe callingSafe = Safe(payable(msg.sender));
+
+        // Check that this guard is enabled on the calling Safe
+        // There is nothing inherently wrong with configuring the guard on a safe that it is not enabled on,
+        // but we choose to revert here to avoid users from mistakenly believing that simply configuring the guard
+        // is enough to activate the delay.
+        if (!_isGuardEnabled(callingSafe)) {
+            revert TimelockGuard_GuardNotEnabled();
+        }
+
+        // Validate timelock delay - must not be longer than 1 year
+        if (_timelockDelay > 365 days) {
+            revert TimelockGuard_InvalidTimelockDelay();
+        }
+
+        // Store the timelock delay for this safe
+        _safeState[callingSafe].timelockDelay = _timelockDelay;
+
+        // Initialize or reset the cancellation threshold to 1
+        _resetCancellationThreshold(callingSafe);
+        emit GuardConfigured(callingSafe, _timelockDelay);
+    }
+
+    /// @notice Schedule a transaction for execution after the timelock delay.
+    /// @dev This function validates signatures in the exact same way as the Safe's own execTransaction function,
+    ///      meaning that the same signatures used to schedule a transaction can be used to execute it later. This
+    ///      maintains compatibility with existing signature generation tools. Owners can use any method to sign the
+    ///      a transaction, including signing with a private key, calling the Safe's approveHash function, or EIP1271
+    ///      contract signatures.
+    /// @param _safe The Safe address to schedule the transaction for.
+    /// @param _nonce The nonce of the Safe for the transaction being scheduled.
+    /// @param _params The parameters of the transaction being scheduled.
+    /// @param _signatures The signatures of the owners who are scheduling the transaction.
+    function scheduleTransaction(
+        Safe _safe,
+        uint256 _nonce,
+        ExecTransactionParams memory _params,
+        bytes memory _signatures
+    )
+        external
+    {
+        // Check that this guard is enabled on the calling Safe
+        if (!_isGuardEnabled(_safe)) {
+            revert TimelockGuard_GuardNotEnabled();
+        }
+
+        // Check that the guard has been configured for the Safe
+        if (_safeState[_safe].timelockDelay == 0) {
+            revert TimelockGuard_GuardNotConfigured();
+        }
+
+        // Get the encoded transaction data as defined in the Safe
+        // The format of the string returned is: "0x1901{domainSeparator}{safeTxHash}"
+        bytes memory txHashData = _safe.encodeTransactionData(
```

Comment:
nit, would postpone defining this until closer to it's initial use at the `checkSignatures` call further down